### PR TITLE
Initial stub for conventions for devices

### PR DESCRIPTION
**Pull request by krwq**
Replaces: https://github.com/dotnet/iot/pull/256

This is an attempt to formalize some conventions we should follow for the devices.

This document should be as short possible so that everyone can read it quickly and later can point to the larger documents with more details.

Please let me know if there is something I should add there.

cc: @tarekgh @JeremyKuhne @joperezr

### CODE REVIEW

File: Documentation/Devices-conventions.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Conventions for devices APIs
+
+## Value returned by sensor
+
+- prefer property (i.e. Temperature) rather than method (i.e. GetTemperature)
```
Comment (tarekgh) [inline Documentation/Devices-conventions.md]: nit: start with upper case letter in every paragraph/line

File: Documentation/Devices-conventions.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+- use `double` when you need to return any floating point value
+- use `Vector2/3/4` for returning vectors`*`
+- value should conform to units conventions (see below)
+- only the most useful APIs should be public, anything else which may be useful but will unlikely get used by most of the people should be protected (inheriting the class allows you to use it but it is not visible by default) or internal/private
```
Comment (tarekgh) [inline Documentation/Devices-conventions.md]: I think we should tell more when protected should be used. like API is believed not going to be used much but there is some specific non main stream scenario can be used so making it protected can allow such scenarios to be achieved.

Comment (krwq) [reply Documentation/Devices-conventions.md]: I'll let someone else fix the wording and merge as is for now (with casing fix)

File: Documentation/Devices-conventions.md (hunk 1 above)
Comment (krwq) [reply Documentation/Devices-conventions.md]: 👍

**Pull request merged**